starter-code/app.js

```diff
@@ -1,5 +1,25 @@
-const express = require('express');
+const express = require("express");
 const app = express();
-const Chuck  = require('chucknorris-io');
+app.set("views", __dirname + "/views");
+app.set("view engine", "ejs");
+const Chuck = require("chucknorris-io");
 const client = new Chuck();
 
+app.get("/random", (req, res, next) => {
+  let joke = {
+    randomJoke: client
+      .getRandomJoke()
+      .then(response => {
+        console.log(response.value);
+      })
+      .catch(err => {
+        // handle error
+      })
+  };
+
+  res.render("index", joke);
```

Comment:
There are two reasons this is not working:

1.  The `render` method expects a js Object as the second argument, so anything you want to pass to the view should be a property of an object. In your case, something like `res.render( "index", { joke: joke } )`.

2.  **More Importantly!** you are trying to pass the joke variable outside of the scope of the callback of an asynchronous request. In other words, you should call the render function inside the callback after the `then`, because that's where you know you already got a response. 

Talk to me for a better explanation in person!